tla/consensus/ccfraft.tla

```diff
@@ -1671,4 +1671,20 @@ DebugCommittedEntriesTermsInv ==
             k <= commitIndex[i]
             => log[i][k].term >= log[j][k].term
 
+
+------------------------------------------------------------------------------
+\* Refinement of the more high-level specification abs.tla that abstracts the
+\* asynchronous network and the message passing between nodes.  Instead, any
+\* node may atomically observe the state of any other node.
+
+MappingToAbs(T, MUC) == 
+  INSTANCE abs WITH
+    Servers <- Servers,
+    Terms <- T,
+    MaxUncommittedCount <- MUC,
+    cLogs <- [i \in Servers |-> [j \in 1..commitIndex[i] |-> log[i][j].term]]
+
+RefinementToAbsProp == \EE T, M : MappingToAbs(T, M)!AbsSpec
```

Comment:
What does \EE do?

Reply:
`\EE` is temporal existential quantification: https://lamport.azurewebsites.net/tla/rhtml/EE.html

Reply:
@lemmy so is it right to understand this as "Spec refines AbsSpec, except for the values of Terms and MaxUncommittedCount"?

Reply:
No, it says that some values for `T` and `M` *exist* (in every state of every behavior) that make the formula `Spec => \EE T, M : MappingToAbs(T, M)!AAbsSpec` true.  In `MCccfraft`, we check that theorem for concrete values of T and MUC.

Background: https://lamport.azurewebsites.net/tla/hiding-and-refinement.pdf